various: {upgrade,drop} GStreamer 0.10 dependency

jtojnar opened the pull request:
###### Motivation for this change
https://github.com/NixOS/nixpkgs/issues/39975

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

jtojnar committed:
firefox: remove GStreamer

Apparently, it has been dropped:

https://bugzilla.mozilla.org/show_bug.cgi?id=1234092

jtojnar committed:
opencv: use GStreamer 1.0

jtojnar committed:
libcanberra: fix GStreamer support

GrahamcOfBorg commented:
Success on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.39985&attempt_id=740c22fc-6d56-4729-a2ef-41f1d5666dd3)

Attempted: opencv

The following builds were skipped because they don't evaluate on x86_64-darwin: libcanberra

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/nslclfc12a613a6dcg307ssg0666kdis-opencv-2.4.13/lib/libopencv_ocl.2.4.13.dylib: fixing dylib
/nix/store/nslclfc12a613a6dcg307ssg0666kdis-opencv-2.4.13/lib/libopencv_photo.2.4.13.dylib: fixing dylib
/nix/store/nslclfc12a613a6dcg307ssg0666kdis-opencv-2.4.13/lib/libopencv_stitching.2.4.13.dylib: fixing dylib
/nix/store/nslclfc12a613a6dcg307ssg0666kdis-opencv-2.4.13/lib/libopencv_superres.2.4.13.dylib: fixing dylib
/nix/store/nslclfc12a613a6dcg307ssg0666kdis-opencv-2.4.13/lib/libopencv_video.2.4.13.dylib: fixing dylib
/nix/store/nslclfc12a613a6dcg307ssg0666kdis-opencv-2.4.13/lib/libopencv_videostab.2.4.13.dylib: fixing dylib
strip is /nix/store/kdff2gim6417493yha769kh00n63lnrw-cctools-binutils-darwin/bin/strip
stripping (with command strip and flags -S) in /nix/store/n6lm79cgs9rzwjm0qdfyd9wb89wjkn3x-opencv-2.4.13-dev/lib
patching script interpreter paths in /nix/store/n6lm79cgs9rzwjm0qdfyd9wb89wjkn3x-opencv-2.4.13-dev
/nix/store/nslclfc12a613a6dcg307ssg0666kdis-opencv-2.4.13
```
</p></details>

GrahamcOfBorg commented:
Failure on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.39985&attempt_id=8948cb6f-0155-4eed-8644-a27934c90b55)

Attempted: libcanberra, opencv

<details><summary>Partial log (click to expand)</summary><p>

```
[ 76%] Building CXX object modules/ocl/CMakeFiles/opencv_perf_ocl.dir/perf/perf_arithm.cpp.o
[ 76%] Building CXX object modules/ocl/CMakeFiles/opencv_perf_ocl.dir/perf/perf_bgfg.cpp.o
[ 77%] Building CXX object modules/ocl/CMakeFiles/opencv_perf_ocl.dir/perf/perf_blend.cpp.o
[ 77%] Building CXX object modules/ocl/CMakeFiles/opencv_perf_ocl.dir/perf/perf_brute_force_matcher.cpp.o
[ 77%] Building CXX object modules/ocl/CMakeFiles/opencv_perf_ocl.dir/perf/perf_calib3d.cpp.o
[ 77%] Building CXX object modules/ocl/CMakeFiles/opencv_perf_ocl.dir/perf/perf_canny.cpp.o
[ 77%] Building CXX object modules/ocl/CMakeFiles/opencv_perf_ocl.dir/perf/perf_color.cpp.o
[ 77%] Building CXX object modules/ocl/CMakeFiles/opencv_perf_ocl.dir/perf/perf_fft.cpp.o
building of '/nix/store/6f2i44g3q2mq5wga94wgpznan9yj5366-opencv-2.4.13.drv' timed out after 3600 seconds
error: build of '/nix/store/6f2i44g3q2mq5wga94wgpznan9yj5366-opencv-2.4.13.drv' failed
```
</p></details>

jtojnar committed:
firefox-bin: remove unneeded GStreamer dependency

jtojnar committed:
qstopmotion: remove unnecessary gstreamer dependency

According to 2.3.1 release notes: “All gstreamer grabber are removed from the application. The reason is the end of the support for gstreamer 0.10 in many distributions.”

GrahamcOfBorg commented:
Success on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.39985&attempt_id=7d7d7f02-edf6-45ee-b9e0-c9b31ef383f8)

Attempted: opencv

The following builds were skipped because they don't evaluate on x86_64-darwin: libcanberra, qstopmotion

<details><summary>Partial log (click to expand)</summary><p>

```
a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowUnsupportedSystem = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowUnsupportedSystem = true; }
to ~/.config/nixpkgs/config.nix.


/nix/store/nslclfc12a613a6dcg307ssg0666kdis-opencv-2.4.13
```
</p></details>

GrahamcOfBorg commented:
Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.39985&attempt_id=bd02ba07-dc1b-4d8a-b3e7-a51d19faa38b)

Attempted: libcanberra, opencv, qstopmotion

<details><summary>Partial log (click to expand)</summary><p>

```
[72/79] Building CXX object CMakeFiles/qstopmotion.dir/src/domain/animation/projectserializer.cpp.o
[73/79] Building CXX object CMakeFiles/qstopmotion.dir/src/frontends/qtfrontend/preferences/transformwidget.cpp.o
[74/79] Building CXX object CMakeFiles/qstopmotion.dir/src/frontends/qtfrontend/preferences/generalwidget.cpp.o
[75/79] Building CXX object CMakeFiles/qstopmotion.dir/src/domain/animation/scene.cpp.o
[76/79] Building CXX object CMakeFiles/qstopmotion.dir/src/frontends/qtfrontend/qtfrontend.cpp.o
[77/79] Building CXX object CMakeFiles/qstopmotion.dir/src/domain/animation/animationproject.cpp.o
[78/79] Building CXX object CMakeFiles/qstopmotion.dir/qstopmotion_autogen/mocs_compilation.cpp.o
ninja: build stopped: subcommand failed.
builder for '/nix/store/09zga2gisdjnmf0gms262kjg5480ib16-qstopmotion-2.3.2.drv' failed with exit code 1
[31;1merror:[0m build of '/nix/store/09zga2gisdjnmf0gms262kjg5480ib16-qstopmotion-2.3.2.drv' failed
```
</p></details>

jtojnar committed:
baresip: switch to GStreamer 1.0

GrahamcOfBorg commented:
Success on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.39985&attempt_id=324e7d7c-676e-42c1-8394-34aa2839f2f2)

Attempted: opencv

The following builds were skipped because they don't evaluate on x86_64-darwin: baresip, libcanberra, qstopmotion

<details><summary>Partial log (click to expand)</summary><p>

```
a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowUnsupportedSystem = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowUnsupportedSystem = true; }
to ~/.config/nixpkgs/config.nix.


/nix/store/nslclfc12a613a6dcg307ssg0666kdis-opencv-2.4.13
```
</p></details>

GrahamcOfBorg commented:
Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.39985&attempt_id=7ade12c8-f896-4b4d-b07a-6a9c5cfebde0)

Attempted: baresip, libcanberra, opencv, qstopmotion

<details><summary>Partial log (click to expand)</summary><p>

```
shrinking /nix/store/ji0fhjxawi5w445mrq7gdprl2499r84y-baresip-0.5.9/lib/baresip/modules/ice.so
shrinking /nix/store/ji0fhjxawi5w445mrq7gdprl2499r84y-baresip-0.5.9/lib/baresip/modules/turn.so
shrinking /nix/store/ji0fhjxawi5w445mrq7gdprl2499r84y-baresip-0.5.9/lib/baresip/modules/stun.so
shrinking /nix/store/ji0fhjxawi5w445mrq7gdprl2499r84y-baresip-0.5.9/lib/baresip/modules/dtmfio.so
shrinking /nix/store/ji0fhjxawi5w445mrq7gdprl2499r84y-baresip-0.5.9/bin/baresip
strip is /nix/store/gp7fylxwn18b7pl2c18ks89hsiaxyfvf-binutils-2.30/bin/strip
stripping (with command strip and flags -S) in /nix/store/ji0fhjxawi5w445mrq7gdprl2499r84y-baresip-0.5.9/lib  /nix/store/ji0fhjxawi5w445mrq7gdprl2499r84y-baresip-0.5.9/bin
patching script interpreter paths in /nix/store/ji0fhjxawi5w445mrq7gdprl2499r84y-baresip-0.5.9
checking for references to /build in /nix/store/ji0fhjxawi5w445mrq7gdprl2499r84y-baresip-0.5.9...
[31;1merror:[0m build of '/nix/store/09zga2gisdjnmf0gms262kjg5480ib16-qstopmotion-2.3.2.drv' failed
```
</p></details>

GrahamcOfBorg commented:
Failure on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.39985&attempt_id=08e7c6f6-246e-4425-b08a-cf84ad909aa6)

Attempted: libcanberra, opencv, qstopmotion

<details><summary>Partial log (click to expand)</summary><p>

```
In file included from /build/source/modules/core/include/opencv2/core/gpumat.hpp:49:0,
                 from /build/source/modules/ts/include/opencv2/ts/gpu_test.hpp:47,
                 from /build/source/modules/nonfree/test/test_precomp.hpp:17,
                 from /build/source/modules/nonfree/test/test_detectors.cpp:43:
/build/source/modules/core/include/opencv2/core/cuda_devptrs.hpp:138:61: note: declared here
         template <typename T> struct __CV_GPU_DEPR_BEFORE__ DevMem2D_ : public PtrStepSz<T>
                                                             ^~~~~~~~~
[ 81%] Building CXX object modules/nonfree/CMakeFiles/opencv_test_nonfree.dir/test/test_features2d.cpp.o
building of '/nix/store/6f2i44g3q2mq5wga94wgpznan9yj5366-opencv-2.4.13.drv' timed out after 3600 seconds
error: build of '/nix/store/6f2i44g3q2mq5wga94wgpznan9yj5366-opencv-2.4.13.drv' failed
```
</p></details>

GrahamcOfBorg commented:
Failure on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.39985&attempt_id=bc7dd385-8575-4550-8f1d-a561525c7f5f)

Attempted: baresip, libcanberra, opencv, qstopmotion

<details><summary>Partial log (click to expand)</summary><p>

```
[ 76%] Building CXX object modules/ocl/CMakeFiles/opencv_perf_ocl.dir/perf/main.cpp.o
[ 76%] Building CXX object modules/ocl/CMakeFiles/opencv_perf_ocl.dir/perf/perf_arithm.cpp.o
[ 76%] Building CXX object modules/ocl/CMakeFiles/opencv_perf_ocl.dir/perf/perf_bgfg.cpp.o
[ 77%] Building CXX object modules/ocl/CMakeFiles/opencv_perf_ocl.dir/perf/perf_blend.cpp.o
[ 77%] Building CXX object modules/ocl/CMakeFiles/opencv_perf_ocl.dir/perf/perf_brute_force_matcher.cpp.o
[ 77%] Building CXX object modules/ocl/CMakeFiles/opencv_perf_ocl.dir/perf/perf_calib3d.cpp.o
[ 77%] Building CXX object modules/ocl/CMakeFiles/opencv_perf_ocl.dir/perf/perf_canny.cpp.o
[ 77%] Building CXX object modules/ocl/CMakeFiles/opencv_perf_ocl.dir/perf/perf_color.cpp.o
building of '/nix/store/6f2i44g3q2mq5wga94wgpznan9yj5366-opencv-2.4.13.drv' timed out after 3600 seconds
error: build of '/nix/store/6f2i44g3q2mq5wga94wgpznan9yj5366-opencv-2.4.13.drv' failed
```
</p></details>
